arch/xtensa/core/xtensa-asm2.c

```diff
@@ -429,6 +449,16 @@ void *xtensa_excint1_c(int *interrupted_stack)
 		_current_cpu->nested = 1;
 	}
 
+#ifdef CONFIG_XTENSA_MMU
+#ifdef CONFIG_USERSPACE
+fixup_out:
+#endif
+	if (is_dblexc) {
+		__asm__ volatile("wsr.depc %0" : : "r"(0));
```

Comment:
The compiler understands that and copy the immediate to a register that is used in the instruction. e.g
```
  0xa00362bb <main+3>     movi.n  a2, 0
  0xa00362bd <main+5>     wsr.depc        a2
```

A double exception can happen because of a page fault during an exception. In the double exception handler we try to load the page from the address that caused the fault and then we `rfde`. But there are certain code that are allowed to fault, e,.g `arch_string_user_nlen`, in these cases the double exception is handled as any other exception and `depc` is cleared out here (and we don't `rfde`). That happens for the next exception/interruption we need to handle we know that it was not a double exception.